tests/notes/class-wc-tests-learn-more-about-variable-product.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * LearnMoreAboutVariableProducts note tests
+ *
+ * @package WooCommerce\Admin\Tests\Notes
+ */
+
+use Automattic\WooCommerce\Admin\Notes\LearnMoreAboutVariableProducts;
+use \Automattic\WooCommerce\Admin\Notes\Notes;
+
+/**
+ * Class WC_Tests_Marketing_Notes
+ */
+class WC_Tests_Learn_More_About_Variable_Product extends WC_Unit_Test_Case {
+
+	/**
+	 * Reset the notes table for each test.
+	 */
+	public function setUp() {
+		parent::setUp();
+		global $wpdb;
+
+		$wpdb->query( "delete from {$wpdb->prefix}wc_admin_notes" );
+	}
+
+	/**
+	 * Tests LearnMoreAboutVariableProducts gets created when a products gets published
+	 */
+	public function test_adding_note_when_product_gets_published() {
+		// Given a new product.
+		$product = array(
+			'post_title'   => 'a product',
+			'post_type'    => 'product',
+			'post_status'  => 'publish',
+			'post_content' => '',
+		);
+
+		// When it is published.
+		wp_insert_post( $product );
+
+		// Then we should have LearnMoreAboutVariableProducts note.
+		$data_store = \WC_Data_Store::load( 'admin-note' );
+		$note_ids   = $data_store->get_notes_with_name( LearnMoreAboutVariableProducts::NOTE_NAME );
+		$this->assertNotEmpty( $note_ids );
+		$this->assertCount( 1, $note_ids );
+
+		$note = Notes::get_note( $note_ids[0] );
+		$this->assertEquals( $note->get_name(), LearnMoreAboutVariableProducts::NOTE_NAME );
+
+		// Adding a second product does not create an additional note.
+		wp_insert_post( $product );
+		$note_ids = $data_store->get_notes_with_name( LearnMoreAboutVariableProducts::NOTE_NAME );
+		$this->assertNotEmpty( $note_ids );
+		$this->assertCount( 1, $note_ids );
+	}
+
+
+	/**
+	 * @dataProvider postProvider
+	 *
+	 * @param string $product product from provider.
+	 */
+	public function test_adding_draft_product_and_non_product_post_does_not_add_note( $product ) {
```

Comment:
Could you add a separate test for creating a `variable` product, and the note not showing?

Also maybe add this in-front of this test: `$wpdb->query( "delete from {$wpdb->prefix}wc_admin_notes" );`
So we know for sure that this test doesn't pass, because of the note already being present in the above test.

Reply:
@louwie17 thank you! I will add a variable test before merging 👍